core/Pos/Core/Block/Union/Instances.hs

```diff
@@ -42,7 +44,9 @@ instance Bi BlockHeader =>
 
 -- | Take 'BlockHeader' from either 'GenesisBlock' or 'MainBlock'.
 getBlockHeader :: Block -> BlockHeader
-getBlockHeader = bimap _gbHeader _gbHeader
+getBlockHeader = \case
+    Left  gb -> BlockHeaderGenesis (_gbHeader gb)
```

Comment:
Next step is to rewrite `Block` from `Either` to custom data type? 🙂

Reply:
Yes :)